### PR TITLE
Fix ticket processing criterias in the personal view

### PR DESCRIPTION
**Pull request by RomainLvr**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !33018

In the personal view, the number of tickets being processed differs from the number of results that appear when you click on the filter.
In this example we are supposed to have 6 tickets in process
image

But in reality, the filter only takes into account tickets in "Progressing (assigned)" status. However, some users apply rules that assign "New" status to certain tickets, even if technicians are assigned to them. In our case, therefore, we only have 4 tickets in the results.
image

One possibility is to modify the filter criteria in this way: 
image

### CODE REVIEW

File: src/Ticket.php [hunk 1]
```diff
@@ -4836,10 +4836,20 @@ public static function showCentralList($start, $status = "process", bool $showgr
...
+
+                        $options['criteria'][7]['link']       = 'OR';
+                        $options['criteria'][7]['criteria'][0]['link']       = 'AND';
+                        $options['criteria'][7]['criteria'][0]['field']      = 12; // status
+                        $options['criteria'][7]['criteria'][0]['searchtype'] = 'equals';
+                        $options['criteria'][7]['criteria'][0]['value']      = 1;
+                        $options['criteria'][7]['criteria'][1]['link']       = 'AND';
+                        $options['criteria'][7]['criteria'][1]['field']      = 5; // users_id_assign
+                        $options['criteria'][7]['criteria'][1]['searchtype'] = 'equals';
+                        $options['criteria'][7]['criteria'][1]['value']      = Session::getLoginUserID();
```
Comment (Rom1-B) [inline src/Ticket.php:4852]: Use the same filter as for the title:

image

**Review by cedric-anne — COMMENTED** [review]
Maybe reusing a condition similar to what is done for groups would be better.
https://github.com/glpi-project/glpi/blob/4a9a451040840fd6f8ece8298aa08318c68cd253/src/Ticket.php#L4752-L4776

`users_id_assign equals Session::getLoginUserID() AND (status equals Ticket::INCOMING OR status equals process)`

**Review by Rom1-B — CHANGES_REQUESTED** [review]

**Comment by AdrienClairembault** [comment]
Looks good to me but the search condition would be clearer this way IMO:

image

File: src/Ticket.php [hunk 2]
```diff
@@ -4831,15 +4831,31 @@ public static function showCentralList($start, $status = "process", bool $showgr
...
-                        $options['criteria'][1]['field']      = 12; // status
-                        $options['criteria'][1]['searchtype'] = 'equals';
-                        $options['criteria'][1]['value']      = 'process';
-                        $options['criteria'][1]['link']       = 'AND';
+                        $options['criteria'] = [
+                            [
+                                'field'        => 5,
+                                'searchtype'   => 'equals',
+                                'value'        => 'myself',
+                                'link'         => 'OR',
```
Comment (AdrienClairembault) [inline src/Ticket.php:4839]: Shouldn't it be `AND` here ?

```suggestion
                                'link'         => 'AND',
```

Comment (RomainLvr) [reply src/Ticket.php]: Yes, thank you :+1:, even if it didn't seem to affect the request.

**Review by AdrienClairembault — APPROVED** [review]

**Review by Rom1-B — APPROVED** [review]

**Review by trasher — APPROVED** [review]

File: src/Ticket.php [hunk 3]
```diff
@@ -4831,15 +4831,31 @@ public static function showCentralList($start, $status = "process", bool $showgr
...
+                                        'link'        => 'AND',
+                                        'field'       => 12,
+                                        'searchtype'  => 'equals',
+                                        'value'       => Ticket::INCOMING,
+                                    ],
+                                    [
+                                        'link'        => 'OR',
+                                        'field'       => 12,
+                                        'searchtype'  => 'equals',
+                                        'value'       => Ticket::ASSIGNED,
```
Comment (cedric-anne) [inline src/Ticket.php:4854]: Why not keeping the `process` value ?
```suggestion
                                        'value'       => 'process',
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**